src/test-unit/query-string.spec.js

```diff
@@ -0,0 +1,114 @@
+import { parse, stringify } from '../lib/query-string';
+
+describe('queryString', () => {
+    it('query string starting with a `?`', () => {
+        expect(parse('?foo=bar')).toEqual({foo: 'bar'});
+    });
+
+    it('query strings starting with a `#`', () => {
+        expect(parse('#foo=bar')).toEqual({foo: 'bar'});
+    });
+
+    it('query strings starting with a `&`', () => {
+        expect(parse('&foo=bar&foo=baz')).toEqual({foo: ['bar', 'baz']});
+    });
+
+    it('parse a query string', () => {
+        expect(parse('foo=bar')).toEqual({foo: 'bar'});
+    });
+
+    it('parse multiple query string', () => {
+        expect(parse('foo=bar&key=val')).toEqual({
+            foo: 'bar',
+            key: 'val'
+        });
+    });
+
+    it('should handle wthiout value', () => {
+        expect(parse('foo')).toEqual({ foo: null });
+        expect(parse('foo&key=val')).toEqual({
+            key: 'val',
+            foo: null
+        });
+
+        expect(parse('foo&foo')).toEqual({foo: [null, null]});
+        expect(parse('foo=&foo')).toEqual({foo: ['', null]});
+    });
+
+    it('handle `+` correctly', () => {
+        expect(parse('foo+faz=bar+baz++')).toEqual({'foo faz': 'bar baz  '});
+    });
+
+    it('handle multiple of the same key', () => {
+        expect(parse('foo=bar&foo=baz')).toEqual({foo: ['bar', 'baz']});
+    });
+
+    it('handle multiple values and preserve appearence order', () => {
+        expect(parse('a=value&a=')).toEqual({a: ['value', '']});
+        expect(parse('a=&a=value')).toEqual({a: ['', 'value']});
+    });
+
+    it('query strings having indexed arrays', () => {
+        expect(parse('foo[0]=bar&foo[1]=baz')).toEqual({
+            'foo[0]': 'bar',
+            'foo[1]': 'baz'
+        });
+    });
+    
```

Comment:
Let’s split the tests for parse and stringify into 2 describe blocks kinda: 
```
describe('queryString', () => {
    describe('parse', () => {
        it('should ...', () => { ... });
        ...
    });
    
    describe('stringify', () => {
        it('should ...', () => { ... });
        ...
    });
});
```

So that the test reports are easier to read and we don't need to repeat the names of the methods in each test name